src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -1338,7 +1309,21 @@ let rec norm : cfg -> env -> stack -> term -> term =
             begin
             match strict_args with
             | None ->
-              let stack = stack |> List.fold_right (fun (a, aq) stack -> Arg (Clos(env, a, BU.mk_ref None, false),aq,t.pos)::stack) args in
+              let stack =
+                List.fold_right
+                  (fun (a, aq) stack ->
+                    let a =
+                      if ((Cfg.cfg_env cfg).erase_erasable_args ||
+                          cfg.steps.for_extraction ||
+                          cfg.debug.erase_erasable_args) //just for experimentation
+                      && U.aqual_is_erasable aq //If we're extracting, then erase erasable arguments eagerly
+                      then U.exp_unit
```

Comment:
This is the bit in the normalizer that erases the argument a to () in case it is marked erasable

Reply:
Note, it is enabled in for_extraction broadly, but also if erase_erasable_args is set, either in the environment (by normalize_for_extraction) or due to a debugging setting.